From 1404d6876f96525f60f865c132774ff21b95e5c8 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Tue, 30 May 2017 00:02:52 +0530
Subject: [PATCH] display: Don't animate screen brightness when turning the
 screen on

Make the display wake from sleep faster by going directly to the target
brightness instead of ramping up to it.

Change-Id: I7172d352f75f6c7b8d93d8c03313492193c854e5
Signed-off-by: Joey Huab <joey@evolution-x.org>
---
 services/core/java/com/android/server/display/RampAnimator.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/display/RampAnimator.java b/services/core/java/com/android/server/display/RampAnimator.java
index 378cdba09520..4894ffa81a4e 100644
--- a/services/core/java/com/android/server/display/RampAnimator.java
+++ b/services/core/java/com/android/server/display/RampAnimator.java
@@ -79,7 +79,7 @@ class RampAnimator<T> {
         final float target = BrightnessUtils.convertLinearToGamma(targetLinear);
 
         // Immediately jump to the target the first time.
-        if (mFirstTime || rate <= 0) {
+        if (mFirstTime || rate <= 0 || mCurrentValue == 0) {
             if (mFirstTime || target != mCurrentValue) {
                 mFirstTime = false;
                 mRate = 0;
-- 
2.43.0

